File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/character/CharacterMinimalInformations.scala





















// Generated on 05/03/2013 10:35:02
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait CharacterMinimalInformations extends AbstractCharacterInformation {
  override val typeId = 110

  protected trait Type extends AbstractCharacterInformation.Type with Serializable {
    override val typeId = CharacterMinimalInformations.typeId

    val level: UByte
    val name: String

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeUByte(level)
      buf.writeString(name)
    }

  }

  protected def create(id: Int, level: UByte, name: String): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val id = buf.readInt()
    val level = buf.readUByte()
    val name = buf.readString()

    create(id, level, name)
  }

}

object CharacterMinimalInformations extends CharacterMinimalInformations {
  case class TypeImpl(id: Int, level: UByte, name: String) extends Type
  override def create(id: Int, level: UByte, name: String) = TypeImpl(id, level, name)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(id: Int, level: UByte, name: String) = create(id, level, name)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/CharacterMinimalInformations.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class CharacterMinimalInformations extends AbstractCharacterInformation {
    public static final short TYPE_ID = 110;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short level;
    public String name;
    
    public CharacterMinimalInformations() { }
    
    public CharacterMinimalInformations(int id, short level, String name) {
        super(id);
        this.level = level;
        this.name = name;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeUByte(level);
        buf.writeString(name);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        level = buf.readUByte();
        if (level < 1 || level > 200)
            throw new RuntimeException("Forbidden value on level = " + level + ", it doesn't respect the following condition : level < 1 || level > 200");
        name = buf.readString();
    }
    
}
